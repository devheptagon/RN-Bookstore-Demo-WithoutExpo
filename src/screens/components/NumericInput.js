import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'
import FlexView from './FlexView'
import RegularView from './RegularView'
import Icon from 'react-native-vector-icons/FontAwesome'
import Theme from '../../constants/Theme'
import { FlexAlign, IconNames } from '../../constants/Enums'

export default class NumericInput extends React.PureComponent {
    state = {
        value: this.props.value || 0
    }

    increase = () => { 
        const { onIncrease } = this.props
        this.setState((state, props) => { 
            const newValue = state.value + 1 
            onIncrease(newValue)
            return { value: newValue} 
        })         
    }

    decrease = () => { 
        const { onDecrease } = this.props
        this.setState((state, props) => { 
            const newValue = state.value - 1 
            onDecrease(newValue)
            return { value: newValue} 
        })         
    }

    render () {                
        return (        
            <RegularView style={styles.Wrapper}>
                <FlexView justifyContent={FlexAlign.SpaceBetween} alignItems={FlexAlign.Center}>
                    <TouchableOpacity onPress={this.increase} style={styles.Button}>
                        <Icon name={IconNames.CaretUp} size={30} />                                         
                    </TouchableOpacity>
                    <Text style={styles.Input}>
                        {this.state.value}
                    </Text>
                    <TouchableOpacity onPress={this.decrease} style={styles.Button}>
                        <Icon name={IconNames.CaretDown} size={30} />                   
                    </TouchableOpacity>
                </FlexView>  
            </RegularView>      
        )
    }
}

NumericInput.propTypes = {
    value: PropTypes.number
}

const styles = StyleSheet.create({
    Wrapper: {        
        paddingTop: 4,
        paddingBottom: 4,
        height: 100
    },
    Input: {
        color: Theme.General.Text.AmountColor,
        fontSize: Theme.General.Text.FontXXBig,
        fontWeight: Theme.General.Text.FontBold
    },
    Button: {
        width: 32, 
        alignItems: FlexAlign.Center
    }
})