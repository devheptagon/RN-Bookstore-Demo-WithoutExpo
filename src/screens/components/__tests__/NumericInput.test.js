import React from "react";
import NumericInput from "../NumericInput";
import renderer from "react-test-renderer";

test("NumericInput renders correctly", () => {
  const tree = renderer.create(<NumericInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
