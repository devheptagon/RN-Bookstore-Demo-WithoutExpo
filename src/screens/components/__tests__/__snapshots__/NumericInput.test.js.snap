// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericInput renders correctly 1`] = `
<View
  style={
    Object {
      "height": 100,
      "paddingBottom": 4,
      "paddingTop": 4,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "opacity": 1,
          "width": 32,
        }
      }
    >
      <Text />
    </View>
    <Text
      style={
        Object {
          "color": "green",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      0
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "opacity": 1,
          "width": 32,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
